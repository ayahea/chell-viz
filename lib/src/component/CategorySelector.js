"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var semantic_ui_react_1 = require("semantic-ui-react");
var CategorySelector = /** @class */ (function (_super) {
    __extends(CategorySelector, _super);
    function CategorySelector(props) {
        return _super.call(this, props) || this;
    }
    CategorySelector.prototype.render = function () {
        return (this.props.categories && (React.createElement(semantic_ui_react_1.Dropdown, { fluid: true, onChange: this.props.onCategoryChange, options: [
                { key: 'all', text: 'all', value: undefined }
            ].concat(this.props.categories.map(function (cat) {
                return { key: cat, text: cat, value: cat };
            })), placeholder: 'Select Category', search: true })));
    };
    return CategorySelector;
}(React.Component));
exports.CategorySelector = CategorySelector;
