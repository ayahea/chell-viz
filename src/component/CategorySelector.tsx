import * as React from 'react';
import { Dropdown } from 'semantic-ui-react';

export interface ICategorySelectorProps {
  categories: string[];
}

export class CategorySelector extends React.Component<ICategorySelectorProps, any> {
  constructor(props: ICategorySelectorProps) {
    super(props);
  }

  public render() {
    return (
      this.props.categories && (
        <Dropdown
          fluid={true}
          options={this.props.categories.map(cat => {
            return { text: cat, value: cat };
          })}
          placeholder={'Select Category'}
          search={true}
        />
      )
    );
  }
}
