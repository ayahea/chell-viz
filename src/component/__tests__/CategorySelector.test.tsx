import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import * as React from 'react';

import { CategorySelector } from '../CategorySelector';

describe('CategorySelector', () => {
  test('Should match existing snapshot when given simple data.', () => {
    const selectorWrapper = shallow(<CategorySelector categories={['Babydoll']} />);
    expect(toJson(selectorWrapper)).toMatchSnapshot();
  });
});
