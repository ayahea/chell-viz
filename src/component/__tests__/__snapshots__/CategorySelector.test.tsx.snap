// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategorySelector Should match existing snapshot when given simple data. 1`] = `
<Dropdown
  additionLabel="Add "
  additionPosition="top"
  closeOnBlur={true}
  deburr={false}
  fluid={true}
  icon="dropdown"
  minCharacters={1}
  noResultsMessage="No results found."
  openOnFocus={true}
  options={
    Array [
      Object {
        "key": "all",
        "text": "all",
        "value": undefined,
      },
      Object {
        "key": "Babydoll",
        "text": "Babydoll",
        "value": "Babydoll",
      },
    ]
  }
  placeholder="Select Category"
  renderLabel={[Function]}
  search={true}
  searchInput="text"
  selectOnBlur={true}
  selectOnNavigation={true}
  wrapSelection={true}
/>
`;
